Align with main consistency document (#3)
chris-wood/draft-group-privacypass-k-check: draft-group-privacypass-k-check.md

    email: jdoe@weylandyutani.com

normative:
  PRIVACYPASS: I-D.ietf-privacypass-architecture
  PRIVACYPASS-ISSUANCE: I-D.ietf-privacypass-protocol
  CONSISTENCY: I-D.ietf-privacypass-key-consistency
  OHTTP: I-D.ietf-ohai-ohttp

informative:
  DOUBLE-CHECK: I-D.schwartz-ohai-consistency-doublecheck

--- abstract

This document describes a protocol called K-Check for implementing HTTP resource consistency checks.
The primary use case for K-Check is for deployments of protocols such as Privacy Pass and Oblivious

> HTTP wherein privacy goals require that clients have a consistent view of some protocol-specific

Reviewer: ```suggestion
HTTP in which privacy goals require that clients have a consistent view of some protocol-specific
```
I personally find "wherein" a bit formal

resource (typically, a public key).

--- middle

# Introduction

Privacy-enhancing protocols such as Privacy Pass {{PRIVACYPASS}} and Oblivious HTTP {{OHTTP}}
require clients to obtain and use a public key for execution. In Privacy Pass,
public keys are used by clients when issuing and redeeming tokens for anonymous
authorization. In Oblivious HTTP (OHTTP), clients use public keys to encrypt
messages to a gateway server.

> Deployments of protocols such as Privacy Pass and OHTTP requires that many, if not all,
> clients share the same key. This is because the privacy properties depend on the client

Reviewer: ```suggestion
Deployments of protocols such as Privacy Pass and OHTTP requires that very large sets of clients
share the same key, or even that all clients globally share the same key. This is because the privacy properties depend on the client
```

anonymity set size. In other words, the key that's used determines the set to which

> a particular belongs. Using a unique, client-speicifc key would yield an anonymity set

Reviewer: ```suggestion
a particular client belongs. Using a unique, client-specific key would yield an anonymity set
```

of size one, therefore violating the desired privacy goals of the system. Clients that

> use the same key are said to have a consistent view of the key.

Reviewer: ```suggestion
use the same key as one another are said to have a consistent view of the key.
```

{{CONSISTENCY}} describes this notion of consistency in more detail. It also outlines
several designs that can be used as the basis for consistency systems. This document is

> a concrete instantiation of one of those designs -- Shared Cache Discovery. In particular,

Reviewer: ```suggestion
a concrete instantiation of one of those designs, "Shared Cache Discovery". In particular,
```

The following terms are used throughout this document:

- Resource: A HTTP resource identified by a URL.
- Normalized resource representation: A unique or otherwise protocol-specific representation
  that is derived from an HTTP resource. The process of normalization is specific to a
  protocol and the resource in question.
- Mirror: A HTTP resource that fetches and caches HTTP resources.

# Mirror Protocol

> The mirror protocol is a simple HTTP-based protocol similar to a reverse-proxy. Each mirror

Reviewer: ```suggestion
The mirror protocol is a simple HTTP-based protocol similar to a reverse proxy. Each mirror
```
